### ISSUE TITLE
Depth labeling is giving wrong contours when pattern is not fully presented

### ISSUE THREAD

**Issue by manuelgitgomes** [lardemua/atom#599]
Hello @miguelriemoliveira and @danifpdra,

As talked to you, I believe I found a bug with the depth labeling.
Be it on automatic or manual labeling, when the pattern is not fully presented in the image, the contours are wrongly calculated, why the points inside the pattern are calculated flawlessly.
As example is shown bellow:

![image](https://github.com/lardemua/atom/assets/58152228/9acf3b90-62a5-4276-870d-de2183eb57bc)

Because the issue is agnostic to the type of labeling, I narrowed it down to the `labelDepthMsg` functions:
https://github.com/lardemua/atom/blob/25aeac8c84fd25c37549b33d557779a71b6bf41b/atom_calibration/src/atom_calibration/collect/label_messages.py#L349

I believe this might be an error on the contour calculation, although I need more debug to be certain.
So, it is either here:

https://github.com/lardemua/atom/blob/25aeac8c84fd25c37549b33d557779a71b6bf41b/atom_calibration/src/atom_calibration/collect/label_messages.py#L558

Or here:
https://github.com/lardemua/atom/blob/25aeac8c84fd25c37549b33d557779a71b6bf41b/atom_calibration/src/atom_calibration/collect/label_messages.py#L566-L588

I will test further. 
If I reach a conclusion, I will document it here.

**Comment by manuelgitgomes** [comment]
I found the source of the error.

After applying Sobel to the pattern mask, the results are as follows:
![image](https://github.com/lardemua/atom/assets/58152228/9d163c72-84ca-47fe-bcc5-ad4abe855e41)
As you can see, the bottom and right side are very misrepresented, therefore the convex hull achieves the following results:
![image](https://github.com/lardemua/atom/assets/58152228/db6cd47a-2050-48be-a988-665e75863820)

To solve this, I though about the Sobel operation.
Is it really needed?

When having a mask of the pattern, the convex hull of the full mask versus the convex hull of the Sobel should ideally be the same right?
This image might help to illustrate my point:
![sobel](https://github.com/lardemua/atom/assets/58152228/d05b5be5-f1ad-4635-a11b-c8f6792c0cab)


Because Sobel is being a source of error, I believe removing Sobel would solve this.
After a small test, it indeed solved. 
However, I only tested in simulated data, therefore I might have missed something important.

What do you think?

**Comment by manuelgitgomes** [comment]
I found the bug!

The Sobel operation, as it was performed, had an error.
In Sobel, the changes from white to black are negative.
Because the image was UInt8, these values were assumed to be 0, erasing these edges.

To solve this, I changed the image to Float64, applied an absolute operation and converted it back to UInt8.
From my testing, it seems to work flawlessly:
![image](https://github.com/lardemua/atom/assets/58152228/b3e62eb9-abda-4293-9f59-4d408e4f38bc)
![image](https://github.com/lardemua/atom/assets/58152228/2543bc3d-dd3f-48e2-b14f-e228f42be380)
